File: frontend/src/services/euAdminSetupService.ts
import axiosInstance from '../utils/axios';

export interface EUAdminSetupStatus {
  hasEUAdmin: boolean;
  canBecomeFirstEUAdmin: boolean;
  canAssignEURegion: boolean;
  isEUAccess: boolean;
  clientIP: string;
  currentUserRegion: string | null;
}

export interface EUAdminSetupResponse {
  success: boolean;
  message?: string;
  user?: any;
}

export const euAdminSetupService = {
  /**
   * Get EU admin setup status
   */
  getSetupStatus: async (): Promise<EUAdminSetupStatus> => {
    try {
      const response = await axiosInstance.get('/eu-admin-setup/status');
      return response.data;
    } catch (error) {
      console.error('Error getting EU admin setup status:', error);
      throw error;
    }
  },

  /**
   * Become the first EU admin
   */
  becomeFirstEUAdmin: async (): Promise<EUAdminSetupResponse> => {
    try {
      const response = await axiosInstance.post('/eu-admin-setup/become-first-eu-admin');
      return response.data;
    } catch (error) {
      console.error('Error becoming first EU admin:', error);
      throw error;
    }
  },

  /**
   * Assign EU region to a user
   */
  assignEURegion: async (userId: number): Promise<EUAdminSetupResponse> => {
    try {
      const response = await axiosInstance.post(`/eu-admin-setup/assign-eu-region/${userId}`);
      return response.data;
    } catch (error) {
      console.error('Error assigning EU region:', error);
      throw error;
    }
  }
};

export default euAdminSetupService;

File: backend/src/main/java/com/ats/service/EUAdminSetupService.java
package com.ats.service;

import com.ats.model.User;
import com.ats.model.Region;

/**
 * Service for EU admin setup logic
 */
public interface EUAdminSetupService {
    
    /**
     * Check if any EU admin already exists in the system
     * @return true if at least one EU admin exists
     */
    boolean hasEUAdmin();
    
    /**
     * Check if the current user can become the first EU admin
     * @param user the user to check
     * @param isEUAccess whether the user is accessing from EU
     * @return true if the user can become the first EU admin
     */
    boolean canBecomeFirstEUAdmin(User user, boolean isEUAccess);
    
    /**
     * Set a user as the first EU admin
     * @param userId the user ID to set as EU admin
     * @param isEUAccess whether the user is accessing from EU
     * @return the updated user
     */
    User setFirstEUAdmin(Long userId, boolean isEUAccess);
    
    /**
     * Check if a user can assign other users to EU region
     * @param user the user to check
     * @return true if the user can assign EU region to others
     */
    boolean canAssignEURegion(User user);
}